Update azidentity's BREAKING_CHANGES.md (#23572)
Azure/azure-sdk-for-go: sdk/azidentity/BREAKING_CHANGES.md

# Breaking Changes

## v1.8.0

### New errors from `NewManagedIdentityCredential` in some environments

> `NewManagedIdentityCredential` now returns an error when `ManagedIdentityCredentialOptions.ID` is set on a platform whose managed identity API doesn't support user-assigned identities. `ManagedIdentityCredential.GetToken()` formerly logged a warning in these cases. Returning an error instead prevents the credential authenticating an unexpected identity, causing a client to act with unexpected privileges. The affected platforms are:

Reviewer: ```suggestion
`NewManagedIdentityCredential` now returns an error when `ManagedIdentityCredentialOptions.ID` is set on a platform whose managed identity API doesn't support user-assigned identities. `ManagedIdentityCredential.GetToken()` formerly logged a warning in these cases. Returning an error instead prevents the credential authenticating an unexpected identity. The affected credentials are:
```
